Sa 3067 get jc system app (#459)
TheJumpCloud/support: PowerShell/JumpCloud Module/Docs/Get-JCSystemApp.md

---
external help file: JumpCloud-help.xml
Module Name: JumpCloud
online version: https://github.com/TheJumpCloud/support/wiki/
schema: 2.0.0
---

# Get-JCSystemApp

## SYNOPSIS

> Returns the System Apps installed to a JumpCloud System(s)

Reviewer: ```suggestion
Returns the applications/programs/linux packages installed on JumpCloud managed system(s). This function queries separate system insights tables to get data for macOS/windows/linux devices.
```

## SYNTAX

### All (Default)
```
Get-JCSystemApp [-SystemID <String>] [-SystemOS <String>] [-SoftwareName <String>] [-SoftwareVersion <String>]
 [<CommonParameters>]
```

### Search
```
Get-JCSystemApp [-SystemID <String>] [-SystemOS <String>] [-SoftwareName <String>] [-SoftwareVersion <String>]
 [-Search] [<CommonParameters>]
```

## DESCRIPTION

> Get-JCSystem app function enables admins to search what software is on their systems

Reviewer: ```suggestion
Get-JCSystem app function helps admins identify what applications/programs or linux packages exist on their JumpCloud managed systems.
```

## EXAMPLES

### Example 1
```powershell

> PS C:\> Get-JCSystemApp -SystemId 'SystemId'

Reviewer: ```suggestion
PS C:\> Get-JCSystemApp -SystemId '6363237ec991136ae59892e4'
```

```

Returns the applications installed in the system with the given -SystemId

### Example 2
```powershell

> PS C:\> Get-JCSystemApp -SystemOs 'MacOs'

Reviewer: ```suggestion
PS C:\> Get-JCSystemApp -SystemOs 'macOS'
```

```

> Returns the 'MacOs' systems and all the applications installed for each system

Reviewer: ```suggestion
Returns the 'macOS' systems and all the applications installed for each system
```

### Example 3
```powershell

> PS C:\> Get-JCSystemApp -SystemOs 'MacOs' -SoftwareName 'JumpCloud-Agent'

Reviewer: ```suggestion
PS C:\> Get-JCSystemApp -SystemOs 'macOS' -SoftwareName 'JumpCloud-Agent'
```

Reviewer: This does not return anything as an example

```

### Example 4
```powershell

> PS C:\> Get-JCSystemApp -SystemOs 'MacOs' -SoftwareName 'JumpCloud-Agent' -SoftwareVersion '1.12.5'

Reviewer: ```suggestion
PS C:\> Get-JCSystemApp -SystemOs 'macOS' -SoftwareName 'JumpCloud-Agent' -SoftwareVersion '1.12.5'
```

```

> Returns the 'MacOs' systems that have a 'JumpCloud Agent' application with the version '1.12.5'

Reviewer: ```suggestion
Returns the 'macOS' systems that have a 'JumpCloud Agent' application with the version '1.12.5'
```

### Example 5
```powershell
PS C:\> Get-JCSystemApp -SoftwareName 'jumpcloud-agent' -Search
```

Returns any 'jumpcloud-agent' software installed in all the os systems

> 

Reviewer: I think we should also call out search by os and id examples here. Another example which would be immedietly helpful for folks would be:

Get-JCSoftwareApp -SystemOS "windows" 

Which would just return all windows programs

## PARAMETERS

### -Search
Global search ex.

> (1.1.2)

Reviewer: parameter does not take input

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: Search
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -SoftwareName
The name of the application you want to search for ex.

> (JumpCloud-Agent, Slack)

Reviewer: This is a decision point all of us need to make before we move forward...

for macOS Systems: `softwareName` refers to the `bundle_name` but we can technically refer to any field in the SI table. Not only do we need to decide which "name" parameter we want to query but we need to document this  so customers know what value will technically return valid results for `softwareName` parameters on windows/ macOS/ linux.

![Screen Shot 2023-01-26 at 10 50 43 AM](https://user-images.githubusercontent.com/54448601/214911161-f381a96a-96a8-4aba-b6f6-9d30d61904ba.png)

Reviewer 2: Based off of what Joe said, I think the `BundleExecutable` or `Name` field would be the likely candidate for finding the desired software app

Author: Decided with filtering with property `name` on the macOs

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -SoftwareVersion
The version of the application you want to search for ex.
(1.1.2)

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -SystemID
The System Id of the JumpCloud system you want to search for applications

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -SystemOS
The type (windows, mac, linux) of the JumpCloud Command you wish to search ex.

> (Windows, Mac, Linux))

Reviewer: ```suggestion
(Windows, macOS, Linux))
```

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

> Accepted values: Windows, MacOs, Linux

Reviewer: ```suggestion
Accepted values: Windows, macOS, Linux
```
